client/homebrew/pages/newPage/newPage.jsx

```diff
@@ -82,238 +73,13 @@ const NewPage = createClass({
-							Not Now
-						</div>
-					</div>
-				</Nav.item>;
-			}
-
-			if(this.state.errors.status == '403' && this.state.errors.response.body.errors[0].reason == 'insufficientPermissions'){
-				return <Nav.item className='save error' icon='fas fa-exclamation-triangle'>
-					Oops!
-					<div className='errorContainer' onClick={this.clearErrors}>
-					Looks like your Google credentials have
-					expired! Visit the log in page to sign out
-					and sign back in with Google
-					to save this to Google Drive!
-						<a target='_blank' rel='noopener noreferrer'
-							href={`https://www.naturalcrit.com/login?redirect=${this.state.url}`}>
-							<div className='confirm'>
-								Sign In
-							</div>
-						</a>
-						<div className='deny'>
-							Not Now
-						</div>
-					</div>
-				</Nav.item>;
-			}
-
-			return <Nav.item className='save error' icon='fas fa-exclamation-triangle'>
-				Oops!
-				<div className='errorContainer'>
-					Looks like there was a problem saving. <br />
-					Report the issue <a target='_blank' rel='noopener noreferrer'
-						href={`https://github.com/naturalcrit/homebrewery/issues/new?body=${encodeURIComponent(errMsg)}`}>
-						here
-					</a>.
-				</div>
-			</Nav.item>;
-		}
-
-		if(this.state.isSaving){
-			return <Nav.item icon='fas fa-spinner fa-spin' className='save'>
-				save...
-			</Nav.item>;
-		} else {
-			return <Nav.item icon='fas fa-save' className='save' onClick={this.save}>
-				save
-			</Nav.item>;
-		}
-	},
-
-	print : function(){
-		window.open('/print?dialog=true&local=print', '_blank');
-	},
-
-	renderLocalPrintButton : function(){
-		return <Nav.item color='purple' icon='far fa-file-pdf' onClick={this.print}>
-			get PDF
-		</Nav.item>;
-	},
-
-	renderNavbar : function(){
-		return <Navbar>
-
-			<Nav.section>
-				<Nav.item className='brewTitle'>{this.state.brew.title}</Nav.item>
-			</Nav.section>
-
-			<Nav.section>
-				{this.renderSaveButton()}
-				{this.renderLocalPrintButton()}
-				<IssueNavItem />
-				<RecentNavItem />
-				<AccountNavItem />
-			</Nav.section>
-		</Navbar>;
-	},
-
 	render : function(){
-		return <div className='newPage sitePage'>
-			{this.renderNavbar()}
-			<div className='content'>
-				<SplitPane onDragFinish={this.handleSplitMove} ref='pane'>
-					<Editor
-						ref='editor'
-						brew={this.state.brew}
-						onTextChange={this.handleTextChange}
-						onStyleChange={this.handleStyleChange}
-						onMetaChange={this.handleMetaChange}
-						renderer={this.state.brew.renderer}
-					/>
-					<BrewRenderer text={this.state.brew.text} style={this.state.brew.style} renderer={this.state.brew.renderer} errors={this.state.htmlErrors}/>
-				</SplitPane>
-			</div>
-		</div>;
+		const googleDriveOptions = [
+			'Set save location to the Homebrewery?',
+			'Set save location to your personal Google Drive storage?'
+		];
+
+		return <EditorPage pageType='new' googleDriveOptions={googleDriveOptions} brew={this.props.brew}></EditorPage>;
```

Comment:
We could do something like this:

```suggestion
return (
	<EditorPage pageType='new' googleDriveOptions={googleDriveOptions} brew={this.props.brew}>
		<NavBar>
			<PrintLink url={this.getPrintLink()}/>
		</NavBar>

	</EditorPage>
);
```

Then inside of `editorPage.jsx` (or perhaps it needs to be inside of `NavBar.jsx`) we can use `{props.children}` wherever we want the insides of `NavBar` to be inserted like so:

```
render : function(){
return <Nav.base>
	<Nav.section>
		<Nav.logo />
		<Nav.item href='/' className='homebrewLogo'>
			<div>The Homebrewery</div>
		</Nav.item>
		<Nav.item>{`v${this.state.ver}`}</Nav.item>
		<PatreonNavItem />
		{this.props.children}   //<======================= HERE
		{/*this.renderChromeWarning()*/}
	</Nav.section>
	{this.props.children}
</Nav.base>;
}
```

See https://reactjs.org/docs/composition-vs-inheritance.html for the details. Might take some finagling but this would get rid of `editorPage` having to know which type the containing page is.

Reply:
Perhaps what we really need is more like this:

return (
	<EditorPage pageType='new' googleDriveOptions={googleDriveOptions} brew={this.props.brew}
		navElements={
			<PrintLink url={this.getPrintLink()}/>
		}
	</EditorPage>
);

And then inside `editorPage` we can  choose where to render `this.props.navElements`.

Reply:
Implemented NavElements as a prop. Currently EditPage passes NavElements while NewPage does not.